tests/test_evaluator_timeout.py

```
@@ -0,0 +1,459 @@
+
+from openevolve.config import EvaluatorConfig
+from openevolve.evaluator import Evaluator
+
+
+class TestEvaluatorTimeout(unittest.TestCase):
+    """Tests for evaluator timeout functionality"""
+
+    def setUp(self):
+        """Set up test evaluation file"""
+        # Create a test evaluation file
+        self.test_eval_file = tempfile.NamedTemporaryFile(mode="w", suffix=".py", delete=False)
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)